Update release-overview

amyblais opened the pull request:
- Adding Cloud release checklist; update on-prem release checklist; clarify release cycle / processes

amyblais committed:
Update release-overview.md

amyblais committed:
Update release-tips.md

amyblais committed:
Update feature-release.md

amyblais committed:
Create cloud-release.md

amyblais commented on operations/research-and-development/product/release-process/cloud-release.md:
```diff
@@ -0,0 +1,30 @@
+- T-minus counts are measured in "working days" (weekdays other than major holidays concurrent in US and Canada) prior to release day.
+
+### A. (T-minus 7 working days) Fast-forward master to cloud branch
+
```
Confirm tickets are resolved.

amyblais committed:
Update cloud-release.md

amyblais committed:
Update cloud-release.md

amyblais committed:
Update feature-release.md

amyblais committed:
Update release-overview.md

amyblais committed:
Update release-overview.md

amyblais commented on operations/research-and-development/product/release-process/cloud-release.md:
```diff
@@ -0,0 +1,31 @@
+    - Fast-forward “cloud” branch to latest master.
+ 
+### B. (T-minus 0 working days) Release day
+
```
Merge docs and changelog
Close the release in Jira

amyblais commented on operations/research-and-development/product/release-process/feature-release.md:
```diff
@@ -280,18 +226,16 @@ The final release is cut - RC cuts and bug fixes should be completed by this dat
-       - The week RC is cut:
-            - [RC Build Testing for core team](https://mattermost.atlassian.net/browse/PLT-2208)
+    - The week RC is cut:
+          - [RC Build Testing for core team](https://mattermost.atlassian.net/browse/PLT-2208)
```
Misaligned

amyblais commented on operations/research-and-development/product/release-process/feature-release.md:
```diff
@@ -303,24 +247,22 @@ The final release is cut - RC cuts and bug fixes should be completed by this dat
+   - Confirm marketing has been posted (screenshots, mail announcement, tweets, blog posts)
+       - Post in the Announcements channel
+   - Update @mattermost Twitter profile with the next release date
+   - Prepare retweet of GitLab release tweet ([see example here](https://community.mattermost.com/core/pl/k7wchwj5mtrhucj6don96yx3sc))
```
Misaligned

amyblais commented on operations/research-and-development/product/release-process/feature-release.md:
```diff
@@ -19,101 +19,68 @@ Notes:
     - Prepare bullet points for release announcement and share for PMs to work on. [Refer to the process here](https://handbook.mattermost.com/operations/messaging-and-math/how-to-guides-for-m-and-m/how-to-create-release-announcements)
+    - Ask PMs and Tech Writer to complete release documentation by T-9 deadline. [Example request](https://community.mattermost.com/core/pl/w4oobh4zpigsfbqskx5ix5jgxo)
+    - Schedule a meeting with PMs and QAs to discuss upcoming features in the next feature release
 3. Dev:
```
Update model/version https://github.com/mattermost/mattermost-server/blob/master/model/version.go#L16

amyblais committed:
Update cloud-release.md

amyblais committed:
Update feature-release.md

justinegeffen commented on operations/research-and-development/product/release-process/cloud-release.md:
```diff
@@ -0,0 +1,33 @@
+# Cloud Release Process
+
+This document outlines the Cloud release process for the Mattermost core team, which draws from what we find works best for us from Agile, Scrum, and Software Development Lifecycle approaches.
```
```suggestion
This document outlines the Mattermost Cloud release process for the Mattermost core team, which draws from what we find works best for us from Agile, Scrum, and Software Development Lifecycle approaches.
```

cwarnermm replied:
```suggestion
This document outlines the Cloud release process for the Mattermost core team. Our release process draws from many best practice Agile, Scrum, and Software Development Lifecycle methodologies.
```

justinegeffen commented on operations/research-and-development/product/release-process/cloud-release.md:
```diff
@@ -0,0 +1,33 @@
+    - Keep track of any regressions on master for fixing.
+    - Add cloud release dates to the Release Google Calendar.
+    - Start working on the changelog (both internal and external).
+    - Check if any translation PRs, pre-packaged plugins, and notice.txt PRs needed.
```
```suggestion
    - Check whether any translation PRs, pre-packaged plugins, and `NOTICE.txt` PRs are needed.
```

cwarnermm commented on operations/research-and-development/product/release-process/cloud-release.md:
```diff
@@ -0,0 +1,33 @@
+## Release Timeline
+
+Notes:
+- All cut-off dates are based on 10am ([San Francisco Time](http://everytimezone.com/)) on the day stated.
```
```suggestion
- All cut-off dates are based on 10am in San Francisco [(Pacific Time](http://everytimezone.com/)) on the day stated.
```

cwarnermm replied:
I updated this section because when someone clicks on the link provided, the site lists times based on time zone rather than by city.

justinegeffen commented on operations/research-and-development/product/release-process/feature-release.md:
```diff
@@ -19,101 +19,69 @@ Notes:
+       - Update download links and testing server links to the latest RCs
+    - After build is cut, tweet announcement that RC1 is ready (see [example](https://community.mattermost.com/core/pl/tefx1ijyz7bs8mabuxmpq9f7pw))
+    - Update https://docs.mattermost.com/administration/open-source-components.html
+    - Ask martin.kraft to provide a list of Database changes, API changes, WebSocket event changes, and `config.json` setting changes for Changelog. [See example request](https://github.com/mattermost/docs/pull/3398#issuecomment-593411501)
```
```suggestion
    - Ask martin.kraft to provide a list of Database changes, API changes, WebSocket event changes, and `config.json` setting changes for changelog. [See example request](https://github.com/mattermost/docs/pull/3398#issuecomment-593411501)
```

justinegeffen commented on operations/research-and-development/product/release-process/feature-release.md:
```diff
@@ -248,28 +194,29 @@ The final release is cut - RC cuts and bug fixes should be completed by this dat
     - Verify all items in the last posted release checklist are complete
-    - Schedule a release retrospective meeting, to be held within five days from the release
     - Update the server upgrade [in-product notice](https://github.com/mattermost/notices)
+    - Schedule a release retrospective meeting, to be held within five days from the release
```
```suggestion
    - Schedule a release retrospective meeting, to be held within five days of the release
```

justinegeffen commented on operations/research-and-development/product/release-process/feature-release.md:
```diff
@@ -280,47 +227,43 @@ The final release is cut - RC cuts and bug fixes should be completed by this dat
     - For the next release, create the following team meetings. If they conflict with existing meetings, check with meeting owner to reschedule or reschedule the release meeting
       - Bug Bash Meeting on the week after Code Complete at 10:00am San Francisco time
-      - Release Triage and Update Meeting each weekday starting at T-15 and ending at T-2 at 9:30am San Francisco time for PM, QA and release dev
+      - Release Triage and Update Meeting each weekday starting at T-20 and ending at T-0 at 9:30am San Francisco time for PM, QA and release dev
```
```suggestion
      - Release Triage and Update Meeting each weekday starting at T-20 and ending at T-0 at 9:30am San Francisco time for PM, QA, and release dev
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
 
+## Overview of Release Cycles
+
+Currently Cloud releases occur on a 2-week cycle, but the goal is to release to Cloud more frequently than every 2 weeks. PRs don’t need to be cherry-picked to releases (both Cloud and on-prem), except for any last bug fixes that get merged after the release branch is cut as well as for any hotfixes.
```
```suggestion
Currently Mattermost Cloud releases occur on a 2-week cycle, but the goal is to release to frequently than every 2 weeks. PRs don’t need to be cherry-picked to releases (both Mattermost Cloud and self-managed), except for any last bug fixes that get merged after the release branch is cut as well as for any hotfixes.
```

cwarnermm replied:
```suggestion
Cloud releases currently occur every 2-weeks, but the goal is to release to Cloud more frequently. Cloud and on-prem PRs don’t need to be cherry-picked to releases, except for any last minute bug fixes that get merged after the release branch is cut, and any necessary hotfixes.
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+
+Currently Cloud releases occur on a 2-week cycle, but the goal is to release to Cloud more frequently than every 2 weeks. PRs don’t need to be cherry-picked to releases (both Cloud and on-prem), except for any last bug fixes that get merged after the release branch is cut as well as for any hotfixes.
+
+Schedule for Cloud releases:
```
```suggestion
Schedule for Mattermost Cloud releases:
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+ - (T-0): Release Day
+
+Schedule for on-prem releases:
+ - (T-30): Feature Review / Judgment Day / Release branch cut
```
```suggestion
 - (T-30): Feature Review/Judgment Day/Release branch cut
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+
+1. The feature is tested on a PR and E2E test automation is added.
+2. Large features are added behind a feature flag.
+3. The feature flag is initially “off” in Cloud and will be rolled out slowly.
```
```suggestion
3. The feature flag is initially “off” in Mattermost Cloud and will be rolled out slowly.
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+2. Large features are added behind a feature flag.
+3. The feature flag is initially “off” in Cloud and will be rolled out slowly.
+    - Feature flag rollout plan: https://developers.mattermost.com/contribute/server/feature-flags/.
+5. On-premise release branch and release candidate are cut based on cloud release. Prior to the cut to on-premise, the Release manager reviews new feature flags and provides a report to the PM/QA team for their review. Essentially a feature will be included in an on-prem release once the feature flag has been removed.
```
```suggestion
5. Self-managed release branch and release candidate are cut based on the Mattermost Cloud release. Prior to the cut to self-managed, the Release Manager reviews new feature flags and provides a report to the PM/QA team for their review. Essentially a feature will be included in a self-managed release once the feature flag has been removed.
```

cwarnermm replied:
```suggestion
5. On-prem release branch and release candidate are cut based on Cloud release. Prior to the cut to on-prem, the Release Manager reviews new feature flags and provides a report to the PM/QA teams for their review. Essentially a feature will be included in an on-prem release once the feature flag has been removed.
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+3. The feature flag is initially “off” in Cloud and will be rolled out slowly.
+    - Feature flag rollout plan: https://developers.mattermost.com/contribute/server/feature-flags/.
+5. On-premise release branch and release candidate are cut based on cloud release. Prior to the cut to on-premise, the Release manager reviews new feature flags and provides a report to the PM/QA team for their review. Essentially a feature will be included in an on-prem release once the feature flag has been removed.
+6. On-premise release candidate is smoke tested.
```
```suggestion
6. Self-managed release candidate is smoke tested.
```

cwarnermm replied:
```suggestion
6. On-prem release candidate is smoke tested.
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+    - Feature flag rollout plan: https://developers.mattermost.com/contribute/server/feature-flags/.
+5. On-premise release branch and release candidate are cut based on cloud release. Prior to the cut to on-premise, the Release manager reviews new feature flags and provides a report to the PM/QA team for their review. Essentially a feature will be included in an on-prem release once the feature flag has been removed.
+6. On-premise release candidate is smoke tested.
+7. On-premise release final is cut and released publicly.
```
```suggestion
7. Self-managed release final is cut and released publicly.
```

cwarnermm replied:
```suggestion
7. On-prem release final is cut and released publicly.
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+
+## Tracking feature flags
+
+Release manager is able to look at the version we have deployed to Cloud. Also, when we are hooked up to split.io there will be a dashboard where you can see active and historic feature flags. Any feature that has its flag removed would be included on any on-prem releases past the point where that removal was merged. More details on feature flags: https://developers.mattermost.com/contribute/serv ...
```
```suggestion
Release Manager is able to look at the version we have deployed to Mattermost Cloud. Also, when we are hooked up to split.io there will be a dashboard where you can see active and historic feature flags. Any feature that has its flag removed would be included on any self-managed releases past the point where that removal was merged. More details on feature flags: https://developers.mattermost.com/contribute/server/feature-flags/.
```

cwarnermm replied:
```suggestion
Release Manager is able to look at the version we have deployed to Cloud. Also, when we are hooked up to split.io, there will be a dashboard where you can see active and historic feature flags. Any feature that has its flag removed would be included on any on-prem releases after that removal was merged. More details on feature flags: https://developers.mattermost.com/contribute/server/feature-flags/.
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+
+## Adding Milestones on PRs and Jira Tickets
+
+Releases are now focused on "shipping features/improvements when they're ready for Cloud, and then they'll get to an on-prem release once they've been on Cloud for 2+ weeks".
```
```suggestion
Releases are now focused on "shipping features/improvements when they're ready for Mattermost Cloud, and then they'll get to a self-managed release once they've been on Mattermost Cloud for 2+ weeks".
```

cwarnermm replied:
```suggestion
Releases are now focused on shipping features and improvements when they're ready for Cloud, and then they'll get to an on-prem release once they've been available on Cloud for 2+ weeks".
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+
+Releases are now focused on "shipping features/improvements when they're ready for Cloud, and then they'll get to an on-prem release once they've been on Cloud for 2+ weeks".
+
+ - A new “cloud” branch (based off from master) is used and any regression bug fixes for the next Cloud release will be cherry-picked there.
```
```suggestion
 - A new “cloud” branch (based off from master) is used and any regression bug fixes for the next Mattermost Cloud release will be cherry-picked there.
```

cwarnermm replied:
```suggestion
 - A new “cloud” branch (based off of master) is used and any regression bug fixes for the next Cloud release will be cherry-picked there.
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+Releases are now focused on "shipping features/improvements when they're ready for Cloud, and then they'll get to an on-prem release once they've been on Cloud for 2+ weeks".
+
+ - A new “cloud” branch (based off from master) is used and any regression bug fixes for the next Cloud release will be cherry-picked there.
+ - This applies to webapp / server / redux / enterprise repos.
```
```suggestion
 - This applies to webapp/server/Redux/Enterprise repos.
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+
+ - A new “cloud” branch (based off from master) is used and any regression bug fixes for the next Cloud release will be cherry-picked there.
+ - This applies to webapp / server / redux / enterprise repos.
+ - A fix version such as “Cloud (November 24)” is added in Jira to track regression bug fixes for Cloud releases.
```
```suggestion
 - A fix version such as “Cloud (November 24)” is added in Jira to track regression bug fixes for Mattermost Cloud releases.
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+ - A new “cloud” branch (based off from master) is used and any regression bug fixes for the next Cloud release will be cherry-picked there.
+ - This applies to webapp / server / redux / enterprise repos.
+ - A fix version such as “Cloud (November 24)” is added in Jira to track regression bug fixes for Cloud releases.
+ - The on-prem releases will be based on Cloud releases. This is decided by the Release Manager / Cloud team based on the timeline between cloud releases and the release day of on-prem releases.
```
```suggestion
 - The self-managed releases will be based on Mattermost Cloud releases. This is decided by the Release Manager/Cloud team based on the timeline between Mattermost Cloud releases and the release day of self-managed releases.
```

cwarnermm replied:
```suggestion
 - The on-prem releases will be based on Cloud releases. This is decided by the Release Manager / Cloud team based on the timeline between Cloud releases and the release day of on-prem releases.
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+ - The on-prem releases will be based on Cloud releases. This is decided by the Release Manager / Cloud team based on the timeline between cloud releases and the release day of on-prem releases.
+ - PRs currently being merged into master don’t need an on-prem milestone added to them. Release Manager will track merged PRs and help ensure that correct milestones are added as needed. Any concerns can be brought up for discussion on a case-by-case basis. There may be a clearer process for this in the future.
+
+## Triaging Cloud customer issues
```
```suggestion
## Triaging Mattermost Cloud Customer Issues
```

cwarnermm replied:
```suggestion
## Triaging Cloud Customer Issues
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+ - Risk and effort of reverting to the last version or fixing a bug
+
+**Criteria**
+1. "We need to revert to the last version" process
```
```suggestion
1. "We need to revert to the last version" process:
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+
+**Criteria**
+1. "We need to revert to the last version" process
+ - Crash or all services are down due to a bug; affects some to all cloud customers.
```
```suggestion
 - Crash or all services are down due to a bug; affects some to all Mattermost Cloud customers.
```

cwarnermm replied:
```suggestion
 - Crash or all services are down due to a bug; affects some to all Cloud customers.
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+**Criteria**
+1. "We need to revert to the last version" process
+ - Crash or all services are down due to a bug; affects some to all cloud customers.
+2. "We need to release this ASAP" process
```
```suggestion
2. "We need to release this ASAP" process:
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+ - Crash or all services are down due to a bug; affects some to all cloud customers.
+2. "We need to release this ASAP" process
+ - A severe regression or loss of function; affects some to all customers.
+3. "It's ok to wait until next release" process
```
```suggestion
3. "It's OK to wait until next release" process:
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+
+**Responders**
+ - Who is making the decision on which process above we need to follow?
+    - In some cases it is the SET on-call commander, in some cases other people such as the release manager or devs who notice / get notified about the report.
```
```suggestion
    - In some cases it is the SET On-Call Commander, in some cases other people such as the Release Manager or devs who notice or get notified about the report.
```

cwarnermm replied:
```suggestion
    - In some cases it is the SET On-Call Commander, in some cases other people such as the Release Manager or developers who notice or get notified about the report.
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+ - Bugs will be fixed by either SET team or respective dev teams, depending on availability and expertise.
+
+**Reports**
+ - The Cloud team has created a central channel for escalations from support (Cloud Support channel in the Staff team). Additionally when a report is posted, it is important to notify the SET lead, release manager, and dev lead of the team that owns the feature.
```
```suggestion
 - The Cloud team has created a central channel for escalations from support (Cloud Support channel in the Staff team). Additionally when a report is posted, it is important to notify the SET lead, Release Manager, and dev lead of the team that owns the feature.
```

cwarnermm replied:
```suggestion
 - The Cloud team has created a central channel for escalations from Support: Cloud Support channel (available in the Staff team). Additionally when a report is posted, it is important to notify the SET Lead, Release Manager, and Development Lead of the team that owns the feature.
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -43,7 +106,7 @@ Mattermost numbers stable releases in the following format:
 **Q: When is release branch cut for a release?**
 
-  - A: One week prior to the release day.
+  - A: Once the cloud release that the next on-prem release is releasing with has been shipped.
```
```suggestion
  - A: Once the Mattermost Cloud release that will be used for the next self-managed release has been shipped.
```

cwarnermm replied:
```suggestion
  - A: Once the Cloud release has been shipped that the next on-prem release is based on.
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -55,11 +118,11 @@ Mattermost numbers stable releases in the following format:
 **Q: What is community.mattermost kept on?**
 
-  - A: community.mattermost is kept on the latest released release.
+  - A: community.mattermost is kept on the most recent cloud release.
```
```suggestion
  - A: community.mattermost is kept on the most recent Mattermost Cloud release.
```

cwarnermm replied:
```suggestion
  - A: community.mattermost is kept on the most recent Cloud release.
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -55,11 +118,11 @@ Mattermost numbers stable releases in the following format:
 **Q: What is community-release.mattermost kept on?**
 
-  - A: community-release.mattermost is kept on the currently in progress release. Once the release branch is cut for a release, then community-release.mattermost is updated to that release branch.
+  - A: community-release.mattermost is kept on the currently in progress cloud release.
```
```suggestion
  - A: community-release.mattermost is kept on the currently in-progress Mattermost Cloud release.
```

cwarnermm replied:
```suggestion
  - A: community-release.mattermost is kept on the currently in-progress Cloud release.
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -89,10 +152,22 @@ Mattermost numbers stable releases in the following format:
+  - There will be some manual RC testing until all release tests are automated.
+
+**Q: Are mobile and desktop included in the cloud first strategy?**
+  - Mobile and desktop apps are not affected at this point.
```
Needs a line break between Q and A.

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -89,10 +152,22 @@ Mattermost numbers stable releases in the following format:
+  - Mobile and desktop apps are not affected at this point.
+
+**Q: Creating tickets for features that were cut from on-prem - Is this also how it continues to be tracked to see whether it's ready for the next release?**
+  - Any feature that's promoted out of the feature flag process would be turned on for the on-prem releases. That way on-prem only gets fully ready or tested features and not ones still being tested. There will be some cases where we need to make a change to fix or change something for the on-prem release (e.g. a bug that only happens if you use a certain on-prem setting or a certain database we  ...
```
Needs a line break between Q and A.

cwarnermm replied:
```suggestion
  - Any feature that's promoted out of the feature flag process would be turned on for the on-prem releases. That way on-prem only gets fully tested features that are ready for release, and does not get the ones still being tested. There will be some cases where we need to make a fix or change something for the on-prem release (e.g., a bug that only occurs if you use a certain on-prem setting or a certain database we don't use in Cloud). In those cases the on-prem release might have some changes that are not a direct copy of the Cloud build.
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -89,10 +152,22 @@ Mattermost numbers stable releases in the following format:
+
+  - There will be some manual RC testing until all release tests are automated.
+
+**Q: Are mobile and desktop included in the cloud first strategy?**
```
```suggestion
**Q: Are mobile and desktop apps included in the "cloud first" strategy?**
```

justinegeffen replied:
I've left this as "cloud" because it's how we refer to this strategy internally.

cwarnermm commented on operations/research-and-development/product/release-process/cloud-release.md:
```diff
@@ -0,0 +1,33 @@
+    - Ask QA to do high priority testing.
+    - Ask Tech Writer to work on docs.
+    - Keep track of any regressions on master for fixing.
+    - Add cloud release dates to the Release Google Calendar.
```
```suggestion
    - Add Cloud release dates to the Release Google Calendar.
```

cwarnermm commented on operations/research-and-development/product/release-process/cloud-release.md:
```diff
@@ -0,0 +1,33 @@
+### A. (T-minus 8 working days) Fast-forward master to cloud branch
+
+1. Release Manager:
+    - Ask Cloud SRE team to update the community-release server to the same version as the cloud test installations.
```
```suggestion
    - Ask Cloud SRE team to update the community-release server to the same version as the Cloud test installations.
```

justinegeffen replied:
```suggestion
    - Ask Cloud SRE team to update the community-release server to the same version as the Mattermost Cloud test installations.
```

cwarnermm commented on operations/research-and-development/product/release-process/cloud-release.md:
```diff
@@ -0,0 +1,33 @@
+    - Post in Announcements channel with a link to the changelog.
+    - Update community servers to the latest version.
+    - Schedule a retrospective.
+    - Confirm PMs are aware of upcoming cloud release dates and feature complete dates.
```
```suggestion
    - Confirm PMs are aware of upcoming Cloud release dates and feature complete dates.
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -89,10 +152,22 @@ Mattermost numbers stable releases in the following format:
+  - The Cloud Releases channel in the Staff team is used for release updates and for posting the changelog. This may be automated in the future. We're also planning to have a dashboard that shows the current version deployed to cloud, and which commits are in.
+
+**Q: How will versioning work?**
+  - Version numbers will be less meaningful for Cloud. Proposing the name of this to be capabilities following the naming used by OpenGL. It's not whether or not a feature is enabled, but it's whether or not the server is capable of supporting the feature.
```
Needs a line break between Q and A.

justinegeffen replied:
```suggestion
  - Version numbers will be less meaningful for Mattermost Cloud. Proposing the name of this to be capabilities following the naming used by OpenGL. It's not whether or not a feature is enabled, but whether or not the server is capable of supporting the feature.
```

justinegeffen commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -89,10 +152,22 @@ Mattermost numbers stable releases in the following format:
+  - Any feature that's promoted out of the feature flag process would be turned on for the on-prem releases. That way on-prem only gets fully ready or tested features and not ones still being tested. There will be some cases where we need to make a change to fix or change something for the on-prem release (e.g. a bug that only happens if you use a certain on-prem setting or a certain database we  ...
+
+**Q: What information does the customer support team need for Cloud releases?**
+  - The Cloud Releases channel in the Staff team is used for release updates and for posting the changelog. This may be automated in the future. We're also planning to have a dashboard that shows the current version deployed to cloud, and which commits are in.
```
Needs a line break between Q and A.

justinegeffen replied:
```suggestion
  - The Cloud Releases channel in the Staff team is used for release updates and for posting the changelog. This may be automated in the future. We're also planning to have a dashboard that shows the current version deployed to Mattermost Cloud, and which commits are in.
```

cwarnermm replied:
```suggestion
  - The Cloud Releases channel in the Staff team is used for release updates and for posting the changelog. This may be automated in the future. We're also planning to have a dashboard that shows the current version deployed to Cloud, and which commits are included.
```

justinegeffen commented on operations/research-and-development/product/release-process/feature-release.md:
```diff
@@ -19,101 +19,69 @@ Notes:
     - Post this checklist in Release Checklist channel
     - Verify all items in the last posted release checklist are complete
     - Cut a Release Candidate [using this process](https://github.com/mattermost/internal-docs/blob/63dcb6d0ff5d20ab9310466065dc15799b12708c/site/content/infra/release-tasks.md) and check CI servers running on release branch
     - Generate a list of contributors for Changelog
```
```suggestion
    - Generate a list of contributors for changelog
```

justinegeffen commented on operations/research-and-development/product/release-process/feature-release.md:
```diff
@@ -128,62 +96,41 @@ Day when Leads and PMs decide which major features are included in the release,
 2. Release Manager:
     - Post this checklist in Release Checklist channel
     - Verify all items in the last posted release checklist are complete
     - Confirm Changelog reflects any changes since it was merged (including known issues and contributors from all repositories)
```
```suggestion
    - Confirm changelog reflects any changes since it was merged (including known issues and contributors from all repositories)
```

justinegeffen commented on operations/research-and-development/product/release-process/feature-release.md:
```diff
@@ -206,7 +153,7 @@ Day after which only **S1 regressions** should be fixed (crashes, security vulne
     - Confirm changes to `config.json` in compatibility section of Changelog are written back to [settings documentation](http://docs.mattermost.com/administration/config-settings.html#configuration-settings)
```
```suggestion
    - Confirm changes to `config.json` in compatibility section of changelog are written back to [settings documentation](http://docs.mattermost.com/administration/config-settings.html#configuration-settings)
```

justinegeffen commented on operations/research-and-development/product/release-process/feature-release.md:
```diff
@@ -215,15 +162,14 @@ The final release is cut - RC cuts and bug fixes should be completed by this dat
     - Close GitHub meta ticket for the release
-    - Add the download links, SHA-256 hash, and GPG signature to the [version archive](http://docs.mattermost.com/administration/upgrade.html#version-archive)
+    - Add the download links, SHA-256 hash, and GPG signature to the [version archive](https://docs.mattermost.com/administration/version-archive.html)
     - Merge Changelog PR after review is complete
```
```suggestion
    - Merge changelog PR after review is complete
```

justinegeffen commented on operations/research-and-development/product/release-process/feature-release.md:
```diff
@@ -215,15 +162,14 @@ The final release is cut - RC cuts and bug fixes should be completed by this dat
-    - Add the download links, SHA-256 hash, and GPG signature to the [version archive](http://docs.mattermost.com/administration/upgrade.html#version-archive)
+    - Add the download links, SHA-256 hash, and GPG signature to the [version archive](https://docs.mattermost.com/administration/version-archive.html)
     - Merge Changelog PR after review is complete
       - If there is a security fix, confirm the Changelog recommends upgrade, with a note mentioning the security level and thanking the security researcher (security process doc for example)
```
```suggestion
      - If there is a security fix, confirm the changelog recommends upgrade, with a note mentioning the security level and thanking the security researcher (security process doc for example)
```

justinegeffen commented on operations/research-and-development/product/release-process/feature-release.md:
```diff
@@ -280,47 +227,43 @@ The final release is cut - RC cuts and bug fixes should be completed by this dat
     - Prepare retweet of GitLab release tweet ([see example here](https://community.mattermost.com/core/pl/k7wchwj5mtrhucj6don96yx3sc))
 2. Docs:
     - Create a new branch on docs for the next release - `vX.X-documentation`
         - Submit a PR for Changelog against the `vX.X-documentation` branch and add a `Work in Progress` label for it
```
```suggestion
        - Submit a PR for changelog against the `vX.X-documentation` branch and add a `Work in Progress` label for it
```

justinegeffen commented on operations/research-and-development/product/release-process/release-tips.md:
```diff
@@ -11,7 +11,7 @@
  - All sections
    - Check that all sections are written and formatted the same way as in previous Changelogs.
```
```suggestion
   - Check that all sections are written and formatted the same way as in previous changelogs.
```

justinegeffen commented on operations/research-and-development/product/release-process/release-tips.md:
```diff
@@ -11,7 +11,7 @@
-   - Work with Dev Ops on this section.
+   - Work with Devs on this section.
  - Next version
    - If the Changelog mentions items regarding upcoming versions, move them to the bottom of the Changelog.
```
```suggestion
   - If the changelog mentions items regarding upcoming versions, move them to the bottom of the changelog.
```

justinegeffen commented on operations/research-and-development/product/release-process/feature-release.md:
```diff
@@ -19,101 +19,69 @@ Notes:
+    - Review the Jira tickets remaining in the current release fix version and push those that won't make it to the next fix version
+    - After mobile release branch is cut, ask dev to cut an RN build
     - Queue a list of MVP candidates in alphabetical order to the MVP Discussion channel [See example](https://community.mattermost.com/core/pl/3its7ifbw7dh58obpasdszf1mr)
     - Draft Changelog in a WIP PR with updates for highlights, feature additions, known issues, compatibility updates for deprecated features, `config.json`, [database changes](https://github.com/mattermost/mattermost-server/blob/master/store/sqlstore/upgrade.go), [API changes](https://github.com/mattermost/mattermost-server/commits/master/model/client.go), [WebSocket event changes](https://githu ...
```
```suggestion
    - Draft changelog in a WIP PR with updates for highlights, feature additions, known issues, compatibility updates for deprecated features, `config.json`, [database changes](https://github.com/mattermost/mattermost-server/blob/master/store/sqlstore/upgrade.go), [API changes](https://github.com/mattermost/mattermost-server/commits/master/model/client.go), [WebSocket event changes](https://github.com/mattermost/mattermost-server/blob/master/model/websocket_message.go#L13), and Go runtime version; [see example](http://docs.mattermost.com/administration/changelog.html#compatibility)
```

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+Currently Cloud releases occur on a 2-week cycle, but the goal is to release to Cloud more frequently than every 2 weeks. PRs don’t need to be cherry-picked to releases (both Cloud and on-prem), except for any last bug fixes that get merged after the release branch is cut as well as for any hotfixes.
+
+Schedule for Cloud releases:
+ - (T-7): 1 week prior to the release day, master is fast-forwarded to the cloud branch.
```
```suggestion
 - (T-7): 1 week prior to the release day, master is fast-forwarded to the “cloud” branch.
```

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+6. On-premise release candidate is smoke tested.
+7. On-premise release final is cut and released publicly.
+
+## Tracking feature flags
```
```suggestion
## Tracking Feature Flags
```

justinegeffen commented:
Thanks, @amyblais, this is incredibly helpful!

Reading these changes I realized I don't know whether we have naming conventions for cloud/Cloud/Mattermost Cloud. So I've changed the mentions of "cloud" to "Mattermost Cloud" but would like @cwarnermm and @adamjclarkson to comment on that. 

Something I noted, which I'm happy to submit in a new PR, is that I think it'd be a good idea to punctuate the lists in the release checklists/processes. They're all consistently unpunctuated but in terms of general consistency I propose punctuating them. LMK what you think and I'll get that done after this is merged. 

I also changed "Changelog" to "changelog" throughout based on how the current changelogs are named although I noticed that the Cloud and Mobile PRs in progress now are both "Changelog". LMK your thoughts on using capital or lowercase. 

Other than that, this is awesome. :)

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+ - This applies to webapp / server / redux / enterprise repos.
+ - A fix version such as “Cloud (November 24)” is added in Jira to track regression bug fixes for Cloud releases.
+ - The on-prem releases will be based on Cloud releases. This is decided by the Release Manager / Cloud team based on the timeline between cloud releases and the release day of on-prem releases.
+ - PRs currently being merged into master don’t need an on-prem milestone added to them. Release Manager will track merged PRs and help ensure that correct milestones are added as needed. Any concerns can be brought up for discussion on a case-by-case basis. There may be a clearer process for this in the future.
```
```suggestion
 - Cloud PRs currently being merged into master don’t need a self-managed release milestone added to them. Release Manager will track merged PRs and help ensure that correct milestones are added as needed. Any concerns can be brought up for discussion on a case-by-case basis. There may be updates to this process in the future.
```

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+1. "We need to revert to the last version" process
+ - Crash or all services are down due to a bug; affects some to all cloud customers.
+2. "We need to release this ASAP" process
+ - A severe regression or loss of function; affects some to all customers.
```
```suggestion
 - A severe regression or loss of functionality; affects some to all Cloud customers.
```

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+2. "We need to release this ASAP" process
+ - A severe regression or loss of function; affects some to all customers.
+3. "It's ok to wait until next release" process
+ - Loss of function, but little effect on customers.
```
```suggestion
 - Loss of function, but little impact on customers.
```

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+**Responders**
+ - Who is making the decision on which process above we need to follow?
+    - In some cases it is the SET on-call commander, in some cases other people such as the release manager or devs who notice / get notified about the report.
+ - Bugs will be fixed by either SET team or respective dev teams, depending on availability and expertise.
```
```suggestion
 - Bugs will be fixed by either the SET team or by respective development teams, depending on availability and expertise.
```

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -89,10 +152,22 @@ Mattermost numbers stable releases in the following format:
+
+  - There will be some manual RC testing until all release tests are automated.
+
+**Q: Are mobile and desktop included in the cloud first strategy?**
```
```suggestion
**Q: Are mobile and desktop updates included in the Cloud-first strategy?**
```

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -89,10 +152,22 @@ Mattermost numbers stable releases in the following format:
+  - There will be some manual RC testing until all release tests are automated.
+
+**Q: Are mobile and desktop included in the cloud first strategy?**
+  - Mobile and desktop apps are not affected at this point.
```
Can we be a bit more clear about whether mobile and desktop updates are included in the cloud-first strategy? The current answer is a bit vague and doesn't directly answer the question.

amyblais replied:
Clarified

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -89,10 +152,22 @@ Mattermost numbers stable releases in the following format:
+**Q: Are mobile and desktop included in the cloud first strategy?**
+  - Mobile and desktop apps are not affected at this point.
+
+**Q: Creating tickets for features that were cut from on-prem - Is this also how it continues to be tracked to see whether it's ready for the next release?**
```
```suggestion
**Q: When creating tickets for features that were cut from on-prem, is this also how it continues to be tracked to see whether it's ready for the next release?**
```

cwarnermm replied:
The question above feels vague --  is there a simpler way to ask this question?

amyblais replied:
Clarified

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -89,10 +152,22 @@ Mattermost numbers stable releases in the following format:
+**Q: Creating tickets for features that were cut from on-prem - Is this also how it continues to be tracked to see whether it's ready for the next release?**
+  - Any feature that's promoted out of the feature flag process would be turned on for the on-prem releases. That way on-prem only gets fully ready or tested features and not ones still being tested. There will be some cases where we need to make a change to fix or change something for the on-prem release (e.g. a bug that only happens if you use a certain on-prem setting or a certain database we  ...
+
+**Q: What information does the customer support team need for Cloud releases?**
```
```suggestion
**Q: What information does the Customer Support team need for Cloud releases?**
```

cwarnermm reviewed (commented):
Great read! Proposed suggestions provided inline.

amyblais commented:
Note to self: add a checklist step to check that tags were created in Github

amyblais committed:
Apply suggestions from code review

Co-authored-by: Justine Geffen <justinegeffen@users.noreply.github.com>
Co-authored-by: Carrie Warner (Mattermost) <74422101+cwarnermm@users.noreply.github.com>

amyblais committed:
Update cloud-release.md

amyblais committed:
Apply suggestions from code review

Co-authored-by: Justine Geffen <justinegeffen@users.noreply.github.com>

amyblais committed:
Apply suggestions from code review

Co-authored-by: Justine Geffen <justinegeffen@users.noreply.github.com>
Co-authored-by: Carrie Warner (Mattermost) <74422101+cwarnermm@users.noreply.github.com>

amyblais committed:
Update release-overview.md

amyblais committed:
Apply suggestions from code review

Co-authored-by: Justine Geffen <justinegeffen@users.noreply.github.com>

amyblais committed:
Update feature-release.md

amyblais committed:
Update release-overview.md

amyblais committed:
Update release-overview.md

amyblais commented:
> Reading these changes I realized I don't know whether we have naming conventions for cloud/Cloud/Mattermost Cloud. So I've changed the mentions of "cloud" to "Mattermost Cloud"

Sounds good!

> 
> Something I noted, which I'm happy to submit in a new PR, is that I think it'd be a good idea to punctuate the lists in the release checklists/processes. They're all consistently unpunctuated but in terms of general consistency I propose punctuating them. LMK what you think and I'll get that done after this is merged.

We can look at that in a separate PR.

> 
> I also changed "Changelog" to "changelog" throughout based on how the current changelogs are named although I noticed that the Cloud and Mobile PRs in progress now are both "Changelog". LMK your thoughts on using capital or lowercase.

Lowercase sounds good!

amyblais committed:
Update feature-release.md

jwilander commented on operations/research-and-development/product/release-process/cloud-release.md:
```diff
@@ -0,0 +1,33 @@
+- All cut-off dates are based on 10am in San Francisco [(Pacific Time](http://everytimezone.com/)) on the day stated.
+- T-minus counts are measured in "working days" (weekdays other than major holidays concurrent in US and Canada) prior to release day.
+
+### A. (T-minus 8 working days) Fast-forward master to cloud branch
```
I know we've been calling it fast-forward but technically it's not and is just a merge 

```suggestion
### A. (T-minus 8 working days) Merge master to the Cloud branch
```

jwilander replied:
Wouldn't the new schedule of cut/merge on Monday and release on Wednesday make this t-minus 9 days?

amyblais replied:
My bad - updated.

jwilander commented on operations/research-and-development/product/release-process/cloud-release.md:
```diff
@@ -0,0 +1,33 @@
+1. Release Manager:
+    - Ask Cloud SRE team to update the community-release server to the same version as the Mattermost Cloud test installations.
+    - Confirm all Jira tickets are resolved for QA testing.
+    - Ask QA to do high priority testing.
```
With the new schedule wouldn't this be on t-minus 8 days? e.g. start testing on the Tuesday and then not release until next Wednesday

amyblais replied:
I clarified the text a bit to "Notify QA that the test servers are ready for QA testing. Normally QA testing starts on the next day (at T-8)."

jwilander commented on operations/research-and-development/product/release-process/cloud-release.md:
```diff
@@ -0,0 +1,33 @@
+    - Start working on the changelog (both internal and external).
+    - Check whether any translation PRs, pre-packaged plugins, and `NOTICE.txt` PRs are needed.
+2. Dev:
+    - Fast-forward “cloud” branch to latest master.
```
```suggestion
    - Merge latest `master` into the Cloud branch
```

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
 
+## Overview of Release Cycles
+
+Currently Mattermost Cloud releases occur on a 2-week cycle, but the goal is to release to frequently. Cloud and Self-Managed PRs don’t need to be cherry-picked to releases, except for any last minute bug fixes that get merged after the release branch is cut, and any necessary hotfixes.
```
```suggestion
Currently Mattermost Cloud releases occur on a 2-week cycle, but the goal is to release more frequently. Cloud and Self-Managed PRs don’t need to be cherry-picked to releases, except for any last minute bug fixes that get merged after the release branch is cut, and any necessary hotfixes.
```

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+Currently Mattermost Cloud releases occur on a 2-week cycle, but the goal is to release to frequently. Cloud and Self-Managed PRs don’t need to be cherry-picked to releases, except for any last minute bug fixes that get merged after the release branch is cut, and any necessary hotfixes.
+
+Schedule for Mattermost Cloud releases:
+ - (T-7): 1 week prior to the release day, master is fast-forwarded to the “cloud” branch.
```
Is this another case of changing "fast-forward" to merge? If so: "...the `master` branch is merged to the Cloud branch."

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+
+Releases are now focused on "shipping features and improvements when they're ready for Mattermost Cloud, and then they'll get to a self-managed release once they've been available on Mattermost Cloud for 2+ weeks".
+
+ - A new “cloud” branch (based off of master) is used and any regression bug fixes for the next Mattermost Cloud release will be cherry-picked there.
```
```suggestion
 - A new Cloud branch (based off of master) is used, and any regression bug fixes for the next Mattermost Cloud release will be cherry-picked there.
```

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+2. "We need to release this ASAP" process:
+ - A severe regression or loss of functionality; affects some to all Cloud customers.
+3. "It's OK to wait until next release" process:
+ - Loss of function, but little impact on customers.
```
```suggestion
 - Loss of function, but little impact on Cloud customers.
```

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+
+**Responders**
+ - Who is making the decision on which process above we need to follow?
+    - In some cases it is the SET On-Call Commander, in some cases other people such as the Release Manager or devs who notice or get notified about the report.
```
```suggestion
    - In some cases it's the SET On-Call Commander, and in some cases it's other people such as the Release Manager or developers who notice or get notified about the report.
```

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+ - Bugs will be fixed by either the SET team or by respective development teams, depending on availability and expertise.
+
+**Reports**
+ - The Cloud team has created a central channel for escalations from Cloud Support channel (available in the Staff team). Additionally when a report is posted, it is important to notify the SET Lead, Release Manager, and dev lead of the team that owns the feature.
```
```suggestion
 - The Cloud team has created a central channel for escalations from Cloud Support channel (available in the Staff team). Additionally when a report is posted, it is important to notify the SET Lead, Release Manager, and Development Lead of the team that owns the feature.
```

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -89,10 +152,26 @@ Mattermost numbers stable releases in the following format:
+
+**Q: How will versioning work?**
+
+  - Version numbers will be less meaningful for Mattermost Cloud. Proposing the name of this to be capabilities following the naming used by OpenGL. It's not whether or not a feature is enabled, but whether or not the server is capable of supporting the feature.
```
```suggestion
  - Version numbers will be less meaningful for Mattermost Cloud. Proposing the name of this to be capabilities following the naming used by OpenGL. With Mattermost Cloud, it's not whether or not a feature is enabled, but whether or not the server is capable of supporting the feature.
```

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+2. Large features are added behind a feature flag.
+3. The feature flag is initially “off” in Mattermost Cloud and will be rolled out slowly.
+    - Feature flag rollout plan: https://developers.mattermost.com/contribute/server/feature-flags/.
+5. Self-managed release branch and release candidate are cut based on the Mattermost Cloud release. Prior to the cut to self-managed, the Release Manager reviews new feature flags and provides a report to the PM/QA teams for their review. Essentially a feature will be included in a self-managed release once the feature flag has been removed.
```
```suggestion
5. Self-Managed release branch and release candidate are cut based on the Mattermost Cloud release. Prior to the cut to Self-Managed, the Release Manager reviews new feature flags and provides a report to the PM/QA teams for their review. Essentially a feature will be included in a Self-Managed release once the feature flag has been removed.
```

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+3. The feature flag is initially “off” in Mattermost Cloud and will be rolled out slowly.
+    - Feature flag rollout plan: https://developers.mattermost.com/contribute/server/feature-flags/.
+5. Self-managed release branch and release candidate are cut based on the Mattermost Cloud release. Prior to the cut to self-managed, the Release Manager reviews new feature flags and provides a report to the PM/QA teams for their review. Essentially a feature will be included in a self-managed release once the feature flag has been removed.
+6. Self-managed release candidate is smoke tested.
```
```suggestion
6. Self-Managed release candidate is smoke tested.
```

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+    - Feature flag rollout plan: https://developers.mattermost.com/contribute/server/feature-flags/.
+5. Self-managed release branch and release candidate are cut based on the Mattermost Cloud release. Prior to the cut to self-managed, the Release Manager reviews new feature flags and provides a report to the PM/QA teams for their review. Essentially a feature will be included in a self-managed release once the feature flag has been removed.
+6. Self-managed release candidate is smoke tested.
+7. Self-managed release final is cut and released publicly.
```
```suggestion
7. Self-Managed release final is cut and released publicly.
```

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+
+## Tracking Feature Flags
+
+Release Manager is able to look at the version we have deployed to Mattermost Cloud. Also, when we are hooked up to split.io, there will be a dashboard where you can see active and historic feature flags. Any feature that has its flag removed would be included on any self-managed releases past the point after that removal was merged. More details on feature flags: https://developers.mattermost.co ...
```
```suggestion
Release Manager is able to look at the version we have deployed to Mattermost Cloud. Also, when we are hooked up to split.io, there will be a dashboard where you can see active and historic feature flags. Any feature that has its flag removed would be included on any Self-Managed releases past the point after that removal was merged. More details on feature flags: https://developers.mattermost.com/contribute/server/feature-flags/.
```

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+
+## Adding Milestones on PRs and Jira Tickets
+
+Releases are now focused on "shipping features and improvements when they're ready for Mattermost Cloud, and then they'll get to a self-managed release once they've been available on Mattermost Cloud for 2+ weeks".
```
```suggestion
Releases are now focused on "shipping features and improvements when they're ready for Mattermost Cloud, and then they'll get to a Self-Managed release once they've been available on Mattermost Cloud for 2+ weeks".
```

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+ - A new “cloud” branch (based off of master) is used and any regression bug fixes for the next Mattermost Cloud release will be cherry-picked there.
+ - This applies to webapp/server/Redux/Enterprise repos.
+ - A fix version such as “Cloud (November 24)” is added in Jira to track regression bug fixes for Mattermost Cloud releases.
+ - The self-managed releases will be based on Mattermost Cloud releases. This is decided by the Release Manager/Cloud team based on the timeline between Mattermost Cloud releases and the release day of self-managed releases.
```
```suggestion
 - The Self-Managed releases will be based on Mattermost Cloud releases. This is decided by the Release Manager/Cloud team based on the timeline between Mattermost Cloud releases and the release day of Self-Managed releases.
```

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -31,6 +33,67 @@ Mattermost numbers stable releases in the following format:
+ - This applies to webapp/server/Redux/Enterprise repos.
+ - A fix version such as “Cloud (November 24)” is added in Jira to track regression bug fixes for Mattermost Cloud releases.
+ - The self-managed releases will be based on Mattermost Cloud releases. This is decided by the Release Manager/Cloud team based on the timeline between Mattermost Cloud releases and the release day of self-managed releases.
+ - Cloud PRs currently being merged into master don’t need a self-managed release milestone added to them. Release Manager will track merged PRs and help ensure that correct milestones are added as needed. Any concerns can be brought up for discussion on a case-by-case basis. There may be updates to this process in the future.
```
```suggestion
 - Cloud PRs currently being merged into master don’t need a Self-Managed release milestone added to them. Release Manager will track merged PRs and help ensure that correct milestones are added as needed. Any concerns can be brought up for discussion on a case-by-case basis. There may be updates to this process in the future.
```

cwarnermm commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -43,7 +106,7 @@ Mattermost numbers stable releases in the following format:
 **Q: When is release branch cut for a release?**
 
-  - A: One week prior to the release day.
+  - A: Once the Mattermost Cloud release that will be used for the next self-managed release has been shipped.
```
```suggestion
  - A: Once the Mattermost Cloud release that will be used for the next Self-Managed release has been shipped.
```

cwarnermm reviewed (dismissed):
Minor proposed suggestions provided inline.

amyblais committed:
Update cloud-release.md

amyblais committed:
Apply suggestions from code review

Co-authored-by: Carrie Warner (Mattermost) <74422101+cwarnermm@users.noreply.github.com>

amyblais committed:
Update release-overview.md

wiersgallak commented on operations/research-and-development/product/release-process/release-overview.md:
```diff
@@ -43,7 +106,7 @@ Mattermost numbers stable releases in the following format:
 **Q: When is release branch cut for a release?**
 
-  - A: One week prior to the release day.
+  - A: Once the Mattermost Cloud release that will be used for the next Self-Managed release has been shipped.
```
Hoping we can get more specific on this one to have a more consistent and obvious connection between Cloud and On-premise releases. Not blocking for now, but this is very vague and we need to be more specific in the future.

amyblais replied:
@wiersgallak I changed the text to: "Self-Managed releases are based off of Mattermost Cloud releases. For example, the Self-Managed v5.32.0 release is based off of the ``cloud-2021-01-26`` release tag. Currently the cadence is that the Mattermost Cloud shipped in the last week of a month will become the next Self-Managed release. The release branch for a Self-Managed release will be cut once the Mattermost Cloud release that will be used for the next Self-Managed release has been shipped."

Would something like this be helpful?

wiersgallak replied:
@amyblais yes, immensely.  Thank you!

wiersgallak reviewed (dismissed):
Non-blocking comment left.

amyblais committed:
Update release-overview.md

amyblais committed:
Update release-overview.md

amyblais committed:
Add cloud release checklist to TOC

cwarnermm reviewed (approved):
Looks great! No changes proposed.
